### PR TITLE
Fix backend basic auth security

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2350

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package org.wso2.choreo.connect.tests.testcases.withapim;
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java:1]: license headers

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        HttpResponse response = HttpsClientRequest.doGet(endpoint, headers);
+        Assert.assertNotNull(response, "Error occurred while invoking the endpoint " + endpoint + " HttpResponse ");
+
+        // test headers
+        Map<String, String> respHeaders = response.getHeaders();
+        Assert.assertTrue(respHeaders.containsKey("authorization"), "Backend did not receive auth header");
+        Assert.assertEquals(respHeaders.get("authorization"), "Basic YWRtaW46W0NAMThlYjgyNjE=",
+                "backend basic auth header is incorrect");
+    }
+}
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java:104]: shall we add new line

**Comment by renuka-fernando** [comment]
From workflow logs
```
[main] INFO 🐳 [docker-compose] - Docker Compose has finished running
Error:  Tests run: 156, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1,610.735 s <<< FAILURE! - in TestSuite
Error:  testBasicAuthSecuredBackend(org.wso2.choreo.connect.tests.testcases.withapim.BackendSecurityTestCase)  Time elapsed: 31.039 s  <<< FAILURE!
java.lang.AssertionError: backend basic auth header is incorrect expected [Basic YWRtaW46W0NAMThlYjgyNjE=] but found [Basic YWRtaW46W0NANDVkYTQxNTk=]
	at org.wso2.choreo.connect.tests.testcases.withapim.BackendSecurityTestCase.testBasicAuthSecuredBackend(BackendSecurityTestCase.java:101)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    BackendSecurityTestCase.testBasicAuthSecuredBackend:101 backend basic auth header is incorrect expected [Basic YWRtaW46W0NAMThlYjgyNjE=] but found [Basic YWRtaW46W0NANDVkYTQxNTk=]
[INFO] 
Error:  Tests run: 156, Failures: 1, Errors: 0, Skipped: 0
```

**Review by pubudu538 — APPROVED** [review]

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                new String[]{}, user, storeRestClient);
+        Utils.delay(TestConstant.DEPLOYMENT_WAIT_TIME * 2, "Interrupted when waiting for the " +
+                "subscription to be deployed");
+
+        //Invoke API
+        Map<String, String> headers = new HashMap<>();
+        headers.put(HttpHeaders.AUTHORIZATION, "Bearer " + accessToken);
+        String endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/1.0.0/echo");
+        HttpResponse response = HttpsClientRequest.doGet(endpoint, headers);
+        Assert.assertNotNull(response, "Error occurred while invoking the endpoint " + endpoint + " HttpResponse ");
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java:110]:
```suggestion
        Assert.assertNotNull(response, "Error occurred while invoking the endpoint " + endpoint);
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/dto/Api.java [hunk 4]
```diff
@@ -20,7 +20,7 @@ public Api(String name, String version, String context, String endpointUrl, Stri
                List<APIOperationsDTO> operationsDTOS, String[] vhosts) {
         this.name = name;
         this.version = version;
-        this.context = context;
+        this.context = endpointUrl;
         this.endpointUrl = context;
         this.tiersCollection = tiersCollection;
```
Comment (suksw) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/dto/Api.java:25]: AFAIK this constructor is not currently used. Which is why we have missed this. How about removing this constructor entirely?

Comment (AmaliMatharaarachchi) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/dto/Api.java:25]: Let's clean it up with another PR.

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211202.25

**Pull request merged**